armory/datasets/preprocessing.py

```diff
@@ -100,3 +107,25 @@ def video_to_canon(
     if resize is not None:
         raise NotImplementedError("resizing video")
     return video
+
+
+def carla_over_obj_det_image(x, modality="rgb"):
+    if modality == "rgb":
+        return image_to_canon(x[0])
+    elif modality == "depth":
+        return image_to_canon(x[1])
+    elif modality == "both":
+        return image_to_canon(tf.concat([x[0], x[1]], axis=-1))
+    else:
+        raise ValueError(
+            f"Found unexpected modality {modality}. Expected one of ('rgb', 'depth', 'both')."
+        )
+
+
+def carla_over_obj_det_dev_label(x, y_object, y_patch_metadata):
+    # convert TF format to PyTorch format of [x1, y1, x2, y2]
+    height, width = x.shape[1:3]  # TODO: better way to do this?
+    converted_boxes = tf.gather(y_object["boxes"], [1, 0, 3, 2], axis=1)
+    converted_boxes *= [width, height, width, height]
+    y_object["boxes"] = converted_boxes
+    return y_object, y_patch_metadata
```

Comment:
This is really hard to follow. Maybe if you start with describing the TF format, then it will make more sense?